app/controllers/V1/searches_controller.rb

```diff
@@ -0,0 +1,112 @@
+module V1
+  class SearchesController < ApplicationController
+    def create
+      data = search_query
+      raw_data = Submission.where(id: data.pluck(:id))
+
+      render json: build_results(data, raw_data), status: :created
+    rescue StandardError => e
+      render json: { message: "AppStore search query raised #{e.message}" }, status: :unprocessable_entity
+    end
+
+  private
+
+    def search_query
+      relation = Search.where(date_submitted: (submitted_from..submitted_to))
+      relation = relation.where(date_updated: (updated_from..updated_to))
+      relation = relation.where(submission_type:) if submission_type
+      relation = relation.where(status:) if status
+      relation = relation.where("has_been_assigned_to ? :caseworker_id", caseworker_id:) if caseworker_id
+      relation = relation.order(**sort_clause)
+
+      relation.where_terms(query)
+    end
+
+    def build_results(data, raw_data)
+      {
+        metadata: {
+          total_results: data.size,
+          page: offset,
+          per_page: limit,
+        },
+        data: data.limit(limit).offset(offset),
+        raw_data: raw_data.limit(limit).offset(offset),
+      }.to_json
+    end
+
+    def submission_type
+      search_params[:submission_type]
+    end
+
+    def submitted_from
+      search_params[:submitted_from]&.to_date
+    end
+
+    def submitted_to
+      search_params[:submitted_to]&.to_date
+    end
+
+    def updated_from
+      search_params[:updated_from]&.to_date
+    end
+
+    def updated_to
+      search_params[:updated_to]&.to_date
+    end
+
+    def status
+      search_params[:status]
+    end
+
+    def caseworker_id
+      search_params[:caseworker_id]
+    end
+
+    def query
+      search_params[:query]
+    end
+
+    def sort_clause
+      return { date_updated: :desc } unless search_params[:sort_by]
+
+      { sort_by => sort_direction }
+    end
+
+    def sort_by
+      search_params.fetch(:sort_by, :date_updated)
```

Comment:
Caseworker is sending `last_updated` rather than `date_updated`

Reply:
so we currently have `date_submitted` and `date_updated` which i quite like the symmetry of and i have found the different naming a useful distinction. This field needs more work anyway (re what date should actually be presented here) which would like to do post this PR in any event. Can we revisit once this is in?